addons/mail/static/src/new/core/message_model.js

```diff
@@ -102,7 +104,7 @@ export class Message {
                 extension: attachment.name.split(".").pop(),
                 originThread: Thread.insert(this._state, attachment.originThread[0][1]),
             })),
-            author: Partner.insert(this._state, data.author),
+            author: data.author ? Partner.insert(this._state, data.author) : this.author,
```

Comment:
Note that, as we talked about this, this is a mistake to store reactive on object.
We should keep storing a raw data (e.g. `partnerId`), and rely on a matching getter to get the corresponding reactive.